### PR TITLE
Fix login message when using Direct Deploy and there is no entity

### PR DESCRIPTION
**Pull request by hatched**
Fixes: #3093

Move the entity loading handling from the Direct Deploy component in the deployment flow to the Deployment Flow component so that the status can be updated cross components.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

- Deploy charms/bundles as usual.
- Use direct deploy on valid charm and bundle urls logged in and not logged in.
- Use direct deploy on an invalid id logged in and not logged in.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7908/

File: jujugui/static/gui/src/app/components/usso-login-link/test-usso-login-link.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ describe('USSOLoginLink', () => {
         displayType={'text'}
         loginToController={sinon.stub()} />);
     const expected = <div className="usso-login">
-      <a className={'logout-link usso-login__action'}
+      <a className={'usso-login__action'}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/usso-login-link/test-usso-login-link.js]: `<a className="usso-login__action"`

File: jujugui/static/gui/src/app/components/usso-login-link/usso-login-link.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ class USSOLoginLink extends React.Component {
   */
   _renderTextLink() {
     return (
-      <a className={'logout-link usso-login__action'}
+      <a className={'usso-login__action'}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/usso-login-link/usso-login-link.js]: `<a className="usso-login__action"`

File: jujugui/static/gui/src/app/components/deployment-flow/test-deployment-flow.js [hunk 3]
```diff
@@ -542,7 +550,8 @@ describe('DeploymentFlow', function() {
         callback={output.props.children[10].props.callback}
         gisf={true}
         isDirectDeploy={false}
-        loginToController={loginToController} />);
+        loginToController={loginToController}
+        showLoginLinks={true}/>);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/test-deployment-flow.js]: `showLoginLinks={true} />);` (space)

File: jujugui/static/gui/src/app/components/deployment-flow/login/test-login.js [hunk 4]
```diff
@@ -251,4 +253,122 @@ describe('DeploymentLogin', function() {
       </div>);
     expect(output).toEqualJSX(expected);
   });
+
+  it('can emit the login links when requested', function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/login/test-login.js]: "emit"? Could be render/display. I usually associate emit with events, but could just be me.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7910/

**Comment by hatched** [comment]
Thanks all for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**